lib/install/management/store_files.go

```diff
@@ -126,7 +126,7 @@ func (d *Dispatcher) deleteDatastoreFiles(ds *object.Datastore, path string, for
 	}
 
 	m := object.NewFileManager(ds.Client())
-	if d.isVSAN(ds) {
+	if d.geVC65() || d.isVSAN(ds) {
```

Comment:
I don't care too much about moving to a consolidated branch, but having this logic here instead of in the pkg/vsphere/datastore code means that every single caller who wants to delete a tree has to do this same version detection or know the specifics of how to do a recursive deletion.